### PR TITLE
Tool guidance

### PR DESCRIPTION
**Pull request by sima-zhu**

### CODE REVIEW

File: tools/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Azure SDK Tools Contribution Guidelines
+
+This is the repository of tool libraries used by Azure SDK purpose.
```
Comment (weshaggard) [inline tools/CONTRIBUTING.md:3]:
```suggestion
This is the repository used to host tools and libraries used by Azure SDK team engineering system. 
```

File: tools/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+For every newly developing tool, create a new folder with name of brief description of the tool. 
+Every tool should serve for single purpose.
+Do not add third party tools directly in your folder, use it as dependencies.
+
+## README
+
+Please add README file for every tool to illustrate:
```
Comment (weshaggard) [inline tools/CONTRIBUTING.md:10]:
```suggestion
Please add README.md file for every tool to illustrate:
```

File: tools/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# Azure SDK Tools Contribution Guidelines
+
+This is the repository of tool libraries used by Azure SDK purpose.
+For every newly developing tool, create a new folder with name of brief description of the tool. 
+Every tool should serve for single purpose.
```
Comment (weshaggard) [inline tools/CONTRIBUTING.md:5]:
```suggestion
```

File: tools/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+Do not add third party tools directly in your folder, use it as dependencies.
+
```
Comment (weshaggard) [inline tools/CONTRIBUTING.md:7]: tools should have entries inside the codeowners file.

File: tools/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+If there is any bundle script, please provide end-to-end test on script as well.
+
+Example: [Custom Test on ci.yml](https://github.com/Azure/azure-sdk-tools/blob/main/tools/code-owners-parser/ci.yml#L35)
+
+## Release ci.yml
+
+For the tool which is publishing to public repository, please provide ci.yml for building, testing and releasing. 
+
+Example: [ci.yml](https://github.com/Azure/azure-sdk-tools/blob/main/tools/CreateRuleFabricBot/ci.yml)
```
Comment (weshaggard) [inline tools/CONTRIBUTING.md:32]: pipelines should use common templates from eng/pipelines/templates.

Please add comments about the naming convention of the pipelines `tools - <tool-name> - ci` for the public builds and `tools - <tool-name>` for internal builds. 

We should also include notes about only including release steps for internal builds, and conditioning those steps similar to https://github.com/Azure/azure-sdk-tools/blob/main/eng/pipelines/templates/stages/archetype-sdk-tool-dotnet.yml#L89.

File: tools/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
+# Azure SDK Tools Contribution Guidelines
```
Comment (weshaggard) [inline tools/CONTRIBUTING.md:1]: It is worth calling out that we allow pretty much any language for the tools.

File: tools/CONTRIBUTING.md (hunk 1 above)
Comment (benbp) [reply tools/CONTRIBUTING.md:3]: This is the repository used to host tools and libraries used by _the_ Azure SDK team engineering system.

File: tools/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+This is the repository of tool libraries used by Azure SDK purpose.
+For every newly developing tool, create a new folder with name of brief description of the tool.
```
Comment (benbp) [inline tools/CONTRIBUTING.md:4]: I would suggest putting this under a new markdown header/section like **Adding a new tool**.

File: tools/CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+This is the repository of tool libraries used by Azure SDK purpose.
+For every newly developing tool, create a new folder with name of brief description of the tool. 
+Every tool should serve for single purpose.
+Do not add third party tools directly in your folder, use it as dependencies.
+
+## README
+
+Please add README file for every tool to illustrate:
+* The purpose of the tool.
+* Prerequsites before use
```
Comment (benbp) [inline tools/CONTRIBUTING.md:12]:
```suggestion
* Prerequisites before use
```

File: tools/CONTRIBUTING.md (hunk 6 above)
Comment (sima-zhu) [reply tools/CONTRIBUTING.md:1]: Added in Line #477

File: tools/CONTRIBUTING.md (hunk 4 above)
Comment (sima-zhu) [reply tools/CONTRIBUTING.md:7]: Add code owner section

File: tools/CONTRIBUTING.md (hunk 5 above)
Comment (sima-zhu) [reply tools/CONTRIBUTING.md:38]: Called out below.

File: tools/CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+Example: [Custom Test on ci.yml](https://github.com/Azure/azure-sdk-tools/blob/main/tools/code-owners-parser/ci.yml#L35)
+
+## Release ci.yml
+
+- For the tool which is publishing to public repository, please provide ci.yml for building, testing and releasing. 
+
+Example: [ci.yml](https://github.com/Azure/azure-sdk-tools/blob/main/tools/CreateRuleFabricBot/ci.yml)
+
+- To publish the tools, use Azure SDK DevOp feeds:
+Example: [Feeds](https://dev.azure.com/azure-sdk/public/_packaging?_a=feed)
```
Comment (weshaggard) [inline tools/CONTRIBUTING.md:42]: This isn't the correct feed and I think we generally should publish to the feed that is most appropriate for the tool we are creating. For example most of the .NET based tools are published to our azure-sdk-for-net feed, but npm based tools will need to go to npm itself. We should probably define those default feeds in our shared publishing templates as we create them.

File: tools/CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
+# Azure SDK Tools Contribution Guidelines
```
Comment (weshaggard) [inline tools/CONTRIBUTING.md:1]: We should put this in the root of the repo.

**Review by weshaggard — APPROVED** [review]
This looks like a good starting point and we can start referring folks to it and update it as more questions arise.

**Pull request merged**